src/PrinterStatus.php

```diff
@@ -0,0 +1,412 @@
+    
+    const BAUD_CODES = [
+        '0000' => 110,
+        '0001' => 300,
+        '0010' => 600,
+        '0011' => 1200,
+        '0100' => 2400,
+        '0101' => 4800,
+        '0110' => 9600,
+        '0111' => 19200,
+        '1000' => 28800,
+        '1001' => 38400,
+        '1010' => 57600,
+        '1011' => 14400,
+    ];
+
+    const PRINT_MODES = [
+        '0' => "Rewind",
+        '1' => "Peel-Off",
+        '2' => "Tear-Off",
+        '3' => "Cutter",
+        '4' => "Applicator",
+        '5' => "Delayed cut",
+        '6' => "Linerless Peel",
+        '7' => "Linerless Rewind",
+        '8' => "Partial Cutter",
+        '9' => "RFID",
+        'K' => "Kiosk",
+        'S' => "Stream",
+    ];
+    
+    /**
+     * This is the parsed ordered list of values from the `~HS` command. Use the `POS_*` constants to determine the
+     * correct index of the value needed.
+     *
+     * e.g.
+     * ```
+     * Array
+     * (
+     *   [0] => 030
+     *   [1] => 0
+     *   [2] => 0
+     *   [3] => 0482
+     *   [4] => 000
+     *   [5] => 0
+     *   //...
+     *```
+     *
+     * @var array
+     */
+    private $data;
+
+    public function __construct($arrayData)
+    {
+        $this->data = $arrayData;
+    }
+
+    /**
+     * Creates a new object from the raw output of the `~HS` command.
+     *
+     * @param string $response
+     * @return PrinterStatus
+     */
+    public static function createFromRawResponse(string $response): self
+    {
+        // Strip out STX ETX characters
+        $response = preg_replace('/[\x02\x03]/','',$response);
+
+        // Split the lines
+        $lines = explode("\r\n", $response);
+
+        // Split the values of each line and merge them together
+        $data = [];
+        foreach ($lines as $line) {
+            $data = array_merge($data, explode(',', $line));
+        }
+
+        return new static($data);
+    }
+
+    /**
+     * Returns the current baud rate of the printer.
+     *
+     * @return int
+     */
+    public function getBaudRate() : int
+    {
+        $code =
+            $this->getCommBit(8) .
+            $this->getCommBit(2) .
+            $this->getCommBit(1) .
+            $this->getCommBit(0);
+
+        return self::BAUD_CODES[$code];
+    }
+
+    /**
+     * Returns the type of handshake the printer is configured for. ('Xon/Xoff' or 'DTR')
+     *
+     * @return string
+     */
+    public function getHandshakeType() : string
+    {
+        return !!$this->getCommBit(7)  ? 'Xon/Xoff' : 'DTR';
+    }
+
+    /**
+     * @return bool
+     */
+    public function isParityEven() : bool
+    {
+        return !!$this->getCommBit(6);
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEnabled() : bool
+    {
+        return !!$this->getCommBit(5);
+    }
+
+    /**
+     * @return int
+     */
+    public function getStopBits() : int
+    {
+        return !!$this->getCommBit(4) ? 1 : 2;
```

Comment:
[NB] Since the documentation references `0` and `1` for these values I think this would be clearer if you referred to the actual values with a switch or some kind of map:

```
switch ($this->getCommBit(4)) {
    case 0: 
        return 2;
    case 1:
        return 1;
}
```

Reply:
Fixed